src/nodes/CodeBlockView.vue

```diff
@@ -0,0 +1,214 @@
+<template>
+	<NodeViewWrapper as="div" :data-mode="viewMode" class="code-block">
+		<div v-if="isEditable" class="code-block-header">
+			<div class="view-switch">
+				<NcActions :aria-label="t('text', 'Code block options')">
+					<NcActionInput :label="t('text', 'Code block language')"
+						:value="type"
+						:show-trailing-button="false"
+						:placeholder="t('text', 'e.g. php, javascript, json, ...')"
+						@input="updateLanguage">
+						<template #icon>
+							<MarkerIcon :size="20" />
+						</template>
+					</NcActionInput>
+
+					<NcActionSeparator v-if="supportPreview" />
+
+					<NcActionButton v-if="supportPreview" :close-after-click="true" @click="viewMode = 'code'">
+						<template #icon>
+							<CodeBraces :size="20" />
+						</template>
+						{{ t('text', 'Source code') }}
+					</NcActionButton>
+					<NcActionButton v-if="supportPreview" :close-after-click="true" @click="viewMode = 'preview'">
+						<template #icon>
+							<Eye :size="20" />
+						</template>
+						{{ t('text', 'Diagram') }}
+					</NcActionButton>
+					<NcActionButton v-if="supportPreview" :close-after-click="true" @click="viewMode = 'side-by-side'">
+						<template #icon>
+							<ViewSplitVertical :size="20" />
+						</template>
+						{{ t('text', 'Both') }}
+					</NcActionButton>
+
+					<NcActionSeparator v-if="supportPreview" />
+
+					<NcActionLink v-if="supportPreview" href="https://mermaid.js.org/intro/" target="_blank">
+						<template #icon>
+							<Help :size="20" />
+						</template>
+						{{ t('text', 'Mermaid documentation') }}
+					</NcActionLink>
+				</NcActions>
+			</div>
+		</div>
+		<div :class="{'split-view': showCode && showPreview }">
+			<pre v-show="showCode" class="split-view__code"><NodeViewContent as="code" :contenteditable="editable" /></pre>
+			<div v-show="showPreview"
+				ref="preview"
+				class="split-view__preview"
+				:contenteditable="false" />
+			<div v-show="false" :id="targetId" :contenteditable="false" />
```

Comment:
I guess the `v-show="false"` is there to initialize the element with `display: none`, right? I wonder whether vue could interfere here later on given that the condition stays `false` all the time :thinking:

Reply:
Ah, now I see - it's never shown. Then nevermind.